backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -1062,4 +1063,138 @@ private String getStringFromLocationList(List<String> locationList) {
+    if (!joinClause.toString().isEmpty()) {
+      finalQueryBuilder.append("SELECT ").append(alias).append(".rootSha ");
+    }
+    finalQueryBuilder.append(rootQueryStringBuilder);
+    finalQueryBuilder.append(joinClause);
+    if (!whereClause.toString().isEmpty()) {
+      finalQueryBuilder.append(" WHERE ").append(whereClause);
+    }
+    finalQueryBuilder.append(orderClause);
+
+    // Build count query
+    StringBuilder countQueryBuilder = new StringBuilder();
+    if (!joinClause.toString().isEmpty()) {
+      countQueryBuilder.append("SELECT COUNT(").append(alias).append(") ");
+    } else {
+      countQueryBuilder.append("SELECT COUNT(*) ");
+    }
+    countQueryBuilder.append(rootQueryStringBuilder);
+    countQueryBuilder.append(joinClause);
+    if (!whereClause.toString().isEmpty()) {
+      countQueryBuilder.append(" WHERE ").append(whereClause);
+    }
+    countQueryBuilder.append(orderClause);
+
+    Query query = session.createQuery(finalQueryBuilder.toString());
+    Query countQuery = session.createQuery(countQueryBuilder.toString());
+    if (!parametersMap.isEmpty()) {
+      parametersMap.forEach(
+          (key, value) -> {
+            if (value instanceof List) {
+              List<Object> objectList = (List<Object>) value;
+              query.setParameterList(key, objectList);
+              countQuery.setParameterList(key, objectList);
+            } else {
+              query.setParameter(key, value);
+              countQuery.setParameter(key, value);
+            }
+          });
+    }
+
+    LOGGER.debug("Final find commit root_sha query : {}", query.getQueryString());
+    if (request.getPageNumber() != 0 && request.getPageLimit() != 0) {
+      // Calculate number of documents to skip
+      int skips = request.getPageLimit() * (request.getPageNumber() - 1);
+      query.setFirstResult(skips);
+      query.setMaxResults(request.getPageLimit());
+    }
+    List<String> resultSet = query.list();
+
+    Map<String, Object> responseMap = new HashMap<>();
+    responseMap.put("result", new HashSet<>(resultSet));
+    responseMap.put("count", countQuery.uniqueResult());
+    return responseMap;
+  }
+
+  @Override
+  public FindRepositoriesBlobs.Response findRepositoriesBlobs(FindRepositoriesBlobs request)
+      throws ModelDBException {
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      session.beginTransaction();
+
+      Map<String, Object> resultSetMap = getRootShaListByCommitsOrRepos(session, request);
+
+      Set<String> rootShaList = (Set<String>) resultSetMap.get("result");
+      Long totalRecords = (Long) resultSetMap.get("count");
+
+      Set<BlobExpanded> blobExpandedSet = new LinkedHashSet<>();
+      for (String rootSha : rootShaList) {
+        Map<String, BlobExpanded> blobExpandedMap =
+            getCommitBlobMap(session, rootSha, request.getLocationPrefixList());
+        blobExpandedSet.addAll(blobExpandedMap.values());
+      }
```

Comment:
this is called in for loop for each commit root_hash so if there will be a multiple repo_ids in a request then this loop iterate for all the commit of the repos.